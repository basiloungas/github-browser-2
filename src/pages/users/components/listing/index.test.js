import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import UsersListing from './index';

import Page from '../../../../components/page';
import UserList from './components/userList';

describe('pages > Users > Listing', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<UsersListing />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('loads content in "Page" with proper "text"', () => {
    const wrapper = shallow(<UsersListing />);

    expect(wrapper.find(Page).prop('header')).toEqual('User list');
  });

  test('display "UserList" component', () => {
    const wrapper = shallow(<UsersListing />);

    expect(wrapper.find(UserList).exists()).toBe(true);
  });
});
