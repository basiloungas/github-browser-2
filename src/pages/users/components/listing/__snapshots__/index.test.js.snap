// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages > Users > Listing renders correctly 1`] = `
<Page
  header="User list"
>
  <NaviLink
    link="/"
    text="Back to homepage"
  />
  <Well
    bsClass="well"
  >
    <withGithubAPIWrapper />
  </Well>
</Page>
`;
